build_test.go

```diff
@@ -270,12 +272,109 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 			Expect(buffer.String()).To(ContainSubstring("Resolving Dotnet Core Runtime version"))
-			Expect(buffer.String()).To(ContainSubstring("Selected dotnet-runtime version (using buildpack.yml): "))
+			Expect(buffer.String()).To(ContainSubstring("Selected dotnet-runtime version (using BP_DOTNET_FRAMEWORK_VERSION): "))
 			Expect(buffer.String()).To(ContainSubstring("Reusing cached layer"))
 			Expect(buffer.String()).NotTo(ContainSubstring("Executing build process"))
 		})
 	})
 
+	context("when version-source of the selected entry is buildpack.yml", func() {
+		it.Before(func() {
+			entryResolver.ResolveCall.Returns.BuildpackPlanEntry = packit.BuildpackPlanEntry{
+				Name: "dotnet-runtime",
+				Metadata: map[string]interface{}{
+					"version-source": "buildpack.yml",
+					"version":        "2.5.x",
+					"launch":         true,
+				},
+			}
+
+			planRefinery.BillOfMaterialCall.Returns.BuildpackPlan = packit.BuildpackPlan{
+				Entries: []packit.BuildpackPlanEntry{
+					{
+						Name: "dotnet-runtime",
+						Metadata: map[string]interface{}{
+							"version-source": "buildpack.yml",
+							"version":        "2.5.x",
+							"launch":         true,
+						},
+					},
+				},
+			}
+		})
+
+		it("prints a deprecation warning", func() {
+			result, err := build(packit.BuildContext{
+				WorkingDir: workingDir,
+				CNBPath:    cnbDir,
+				Stack:      "some-stack",
+				BuildpackInfo: packit.BuildpackInfo{
+					Name:    "Some Buildpack",
+					Version: "0.1.2",
+				},
+				Plan: packit.BuildpackPlan{
+					Entries: []packit.BuildpackPlanEntry{
+						{
+							Name: "dotnet-runtime",
+							Metadata: map[string]interface{}{
+								"version-source": "buildpack.yml",
+								"version":        "2.5.x",
+								"launch":         true,
+							},
+						},
+					},
+				},
+				Layers: packit.Layers{Path: layersDir},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(result).To(Equal(packit.BuildResult{
+				Plan: packit.BuildpackPlan{
+					Entries: []packit.BuildpackPlanEntry{
+						{
+							Name: "dotnet-runtime",
+							Metadata: map[string]interface{}{
+								"version-source": "buildpack.yml",
+								"version":        "2.5.x",
+								"launch":         true,
+							},
+						},
+					},
+				},
+				Layers: []packit.Layer{
+					{
+						Name: "dotnet-core-runtime",
+						Path: filepath.Join(layersDir, "dotnet-core-runtime"),
+						SharedEnv: packit.Environment{
+							"DOTNET_ROOT.override": filepath.Join(workingDir, ".dotnet_root"),
+						},
+						BuildEnv: packit.Environment{
+							"RUNTIME_VERSION.override": "2.5.x",
+						},
+						LaunchEnv:        packit.Environment{},
+						ProcessLaunchEnv: map[string]packit.Environment{},
+						Build:            false,
+						Launch:           true,
+						Cache:            false,
+						Metadata: map[string]interface{}{
+							"dependency-sha": "some-sha",
+							"built_at":       timeStamp.Format(time.RFC3339Nano),
+						},
+					},
+				},
+			}))
```

Comment:
For sake of brevity, I don't think it's necessary to assert on the build result in this test. We already test that a given selected plan entry is correctly returned in the build result.